### PR TITLE
feat: update dependencies to include numpy 1.25.x and Pennylane 0.33.x

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
+numpy==1.25.2
+pandas==2.1.3
+pennylane==0.33.1
+PennyLane-Lightning==0.33.1
+qiskit-aer==0.13.0
 qiskit_braket_provider==0.0.4
-qiskit-terra==0.25.2.1
-scipy==1.11.3
+qiskit-terra==0.45.0
+scipy==1.9.3
```
Comment (math411) [inline requirements.txt:28]: Downgrading as `openfermion 1.6.0 depends on scipy<1.10.0 and >=1.1.0`.

**Comment by math411** [comment]
https://github.com/scipy/scipy/issues/19344

scipy calls a numpy test library - this causes issues for the jobs example. Once openfermion can support a later version of scipy, Braket should look into later support for this library.

**Comment by math411** [comment]
The build failures are expected currently. Waiting on a new scipy release to resolve and get this merged.

**Comment by math411** [comment]
As a result of patching `stdout` we are causing numpy issues which attempts to read directly there. stdout.output is now a Mock object and not iterable. These changes work to make it such that the old behavior returns

**Comment by speller26** [comment]
I wouldn't call this `infra` because this actually changes what users will get in their environments.

**Review by speller26 — APPROVED** [review]

**Pull request merged**